tests/e2e/k8s/suite.go

```diff
@@ -0,0 +1,190 @@
+// Copyright 2023 The ClusterLink Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/stretchr/testify/suite"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/clusterlink-net/clusterlink/pkg/bootstrap/platform"
+	"github.com/clusterlink-net/clusterlink/tests/e2e/k8s/util"
+)
+
+const (
+	clusterCount     = 3
+	exportedLogsPath = "/tmp/clusterlink-k8s-tests"
+)
+
+var images = [...]string{"cl-controlplane", "cl-dataplane", "cl-go-dataplane"}
+
+var iperfService = util.Service{
```

Comment:
The images are of course relevant to suite setup.
The services are defined here since I expect them to be re-used across tests.

Reply:
Agree regarding the clusterlink images.
You can still share the service definitions without making them part of the suite file (e.g., defaultservices.go)

Reply:
For now, the suite file (and the number of global services) is small.
I say we keep it as-is for now, until we see (if at all) the number of global services grows much larger.

This package is already a bit confusing since it mixes 3 types of files:
1. suite.go (defining the suite)
2. k8s_test.go (defining the go test)
3. other files which contain actual tests

Adding further files IMO makes it harder to get around.